timezone/src/modules/Timezone.rb

```diff
@@ -984,6 +986,23 @@ def Summary
       HTML.List(ret)
     end
 
+
+    # Checks whether the system has Windows installed on a primary partition with NTFS
+    def system_has_windows?
+      storage = Y2Storage::StorageManager.instance
+      Storage::Ntfs::all(storage.probed).each do |ntfs|
+        ntfs.blk_devices.each do |blk_device|
+          if Storage::partition?(blk_device)
+            if Storage::to_partition(blk_device).type == Storage::PartitionType_PRIMARY
+              return true if ntfs.detect_content_info.windows?
+            end
+          end
+        end
```

Comment:
I believe that using that using `Y2Storage::Refinements::DevicegraphLists` the code can be rewritten as:
```ruby
def system_has_windows?
  devicegraph = Y2Storage::StorageManager.instance.probed
  prim_parts = devicegraph.partitions.with(type: Storage::PartitionType_PRIMARY)
  prim_parts.filesystems.with(type: Storage::FsType_NTFS).any? { |fs| fs.detect_content_info.windows? }
end
```